# Mapping RNA-seq data

The following steps require the raw RNA-seq fastq files as input to generate _RAWcounts_ and TPM _abundance_
files as output.
The following steps were performed on each of the fastq files.

## Software requirements

|Software|Version|
| --- | --- |
|BBmap|38.26|
|Hisat2|2.1.0|
|Samtools|1.9|
|Kallisto|0.43.0|
|python|2.7.13|
|HTseq|0.9.1|

## Trim adaptors from raw reads

To reove the adaptors from sequencing reads, the _ADAPTERS_ file containing sequencing adapters was used. 

```
bbduk.sh in=<FASTQ> out=<FASTQTRIM> minlen=20 ref=<ADAPTERS> ktrim=14 minkmerhits=7 mink=11 hdist=1 stats=<STATSFILE>
```

## Map to reference genome

A previously hisat2 indexed genome _INDEX_ was used as reference.

```bash
hisat2 -q --dta --rna-strandness R -k 12 --summary-file <OUTSUMFILE> --met-file <METFILE> --no-unal -x <INDEX> -U <FASTQTRIM> -S <OUTsam>

samtools sort -o <OUTsortedbam> -T <TMP> <OUTsam>

samtools view -bq 5 <OUTsortedbam> <OUTuniqbam>
```

## Calculate raw counts

The _GRCz11GTF_ corresponds to a gtf file from Ensembl release 92.

```bash
htseq-count -f <OUTuniqbam> -r pos -s reverse -t exon -i gene_id -m intersection-nonempty $bam <GRCz11GTF> > <RAWcounts>
```

## Calculate TPM per gene

A previouly kallisto indexed transcriptome file _KALindx_, containing the annotated transcripts sequences from Ensembl 92 release, was used.

```bash
kallisto quant -i <KALindx> -o <OUTDIR> -b 100 --rf-stranded --single -l 290 -s 20 <FASTQTRIM>
```
