## Generate metage profiles from DEGs

The aligned BAM files are used to generate number of reads on each position

## Software requirements

## Convert BAM to BedGraph

```bash
genomeCoverageBed -ibam <OUTsortbam> -d -split > <OUTbg>
bed12ToBed6 -i <BED> > <OUTtmpbed>
coverageBed -split -a <OUTtmpbed> -b <OUTsortbam>  -d  > <OUT6bed>
```

## Plot metagene profiles

```r
args = commandArgs(TRUE)
i = args[1]
j = args[2]
k = args[3]
l = args[4]
m = args[5]
n = args[6]
o = args[7]
p = args[8]
maindir = args[9]

library(dplyr)
library(ggplot2)
library(data.table)

bedfiles<-unlist(lapply(strsplit(list.files(Sys.glob(file.path(maindir, 'MetagenesBED6', i, j, k, l, o, 'TPs', m, n, 'Coverage', '*', paste0('*_', m), '*')), pattern = p), '.', fixed = T), `[`, 1))
bedfiles<-unlist(lapply(strsplit(basename(bedfiles), '.', fixed = T), `[`, 1))
FinalTab <- c()
for(q in bedfiles){
	tmpbed<-read.table(Sys.glob(file.path(maindir, 'MetagenesBED6', i, j, k, l, o, 'TPs', m, n, 'Coverage', '*/*/*', paste0(q, '.bed'))), sep = '\t', stringsAsFactors = F, header = F)
	colnames(tmpbed)<-c("Chr", "Start", "Stop", "Name", "Feature", "Strand", "Pos", "Counts")
	tmpbed<-data.table(tmpbed)
	tmpbed$Pos<-unlist(Map(':', 1, (tmpbed[, lapply(.SD, length), by = Name, .SDcols = 8])$Counts))
	binned<-tmpbed %>% group_by(Name) %>% mutate(BIN = cut_number(Pos, 100, seq(1,100))) %>% ungroup() %>% group_by(BIN) %>% summarise_at("Counts", mean) %>% ungroup()
	binned$rep <- rep(q)
	if(grepl("WT", q)){
		binned$GT<-rep('WT')
	}else{
		binned$GT<-rep('Ski7')
	}
	FinalTab <- rbind(FinalTab, binned)
}
maxlim<-round(max(FinalTab$Counts))+1
x=ggplot(ungroup(FinalTab), aes(as.double(BIN), Counts, group = rep, col = GT)) + geom_line(stat = "identity") + scale_y_continuous(limits = c(0,maxlim)) + labs(title = paste(m, "in", i, k), x="% length")
dir.create(paste(maindir, 'MetagenesBED6', i, j, k, l, o, 'TPs', m, n, "Graphs", sep = '/'), recursive = T)
dir.create(paste(maindir, 'MetagenesBED6', i, j, k, l, o, 'TPs', m, n, "Tables", sep = '/'), recursive = T)
NAME=paste0(paste(maindir, 'MetagenesBED6', i, j, k, l, o, 'TPs', m, n, "Graphs", sep = '/'), "/", p, "_", i, "_", j, "_", k, "_", l, "_", m, "_", n, "_Metagene_Absolute.pdf")
ggsave(NAME, x)
write.table(FinalTab, paste0(paste(maindir, 'MetagenesBED6', i, j, k, l, o, 'TPs', m, n, "Tables", sep = '/'), "/", p, "_", i, "_", j, "_", k, "_", l, "_", m, "_", n, "_Metagene_Absolute.txt"), sep = '\t', quote = F, row.names = F)

maxes<-by(FinalTab$Counts, FinalTab$rep, max)
maxes<-rep(maxes, each = 100)
FinalTab$Counts <- FinalTab$Counts/maxes
x=ggplot(FinalTab, aes(as.double(BIN), Counts, group = rep, col = GT)) + geom_line(stat = "identity") + labs(title = paste(j, "in", i, k, l, m), x="% length")
NAME=paste0(paste(maindir, 'MetagenesBED6', i, j, k, l, o, 'TPs', m, n, "Graphs", sep = '/'), "/", p, "_", i, "_", j, "_", k, "_", l, "_", m, "_", n, "_Metagene_Relative.pdf")
ggsave(NAME, x)
write.table(FinalTab, paste0(paste(maindir, 'MetagenesBED6', i, j, k, l, o, 'TPs', m, n, "Tables", sep = '/'), "/", p, "_", i, "_", j, "_", k, "_", l, "_", m, "_", n, "_Metagene_Relative.txt"), sep = '\t', quote = F, row.names = F)
```
