## Generate metage profiles from DEGs

The aligned BAM files are used to generate number of reads on each position

## Software requirements

| Software | Version |
|---|---|
| Bedtools | 2.27.1 |

## Convert BAM to BedGraph

As a reference _BED_ file, use a gtf file converted to BED but keeping the gene_IDs

```bash
bed12ToBed6 -i <BED> > <OUTtmpbed>
coverageBed -split -a <OUTtmpbed> -b <OUTsortbam>  -d  > <OUT6bed>
```

## Plot metagene profiles

Required: BED files separated into Up, Down, Unchanged genes

```r
i = # Oocytes, Eggs, or Embryos
j = # Time point to be plotted
k = # Up, Down, or Unchanged


maindir = "DEGs/"

library(dplyr)
library(ggplot2)
library(data.table)

# Read bedfiles
bedfiles<-unlist(lapply(strsplit(list.files(Sys.glob(file.path(maindir, 'MetagenesBED6', i, j)), pattern = k), '.', fixed = T), `[`, 1))
bedfiles<-unlist(lapply(strsplit(basename(bedfiles), '.', fixed = T), `[`, 1))
FinalTab <- c()
for(l in bedfiles){
	tmpbed<-read.table(Sys.glob(file.path(maindir, 'MetagenesBED6', i, j, paste0(l, '.bed'))), sep = '\t', stringsAsFactors = F, header = F)
	colnames(tmpbed)<-c("Chr", "Start", "Stop", "Name", "Feature", "Strand", "Pos", "Counts")
	tmpbed<-data.table(tmpbed)
	tmpbed$Pos<-unlist(Map(':', 1, (tmpbed[, lapply(.SD, length), by = Name, .SDcols = 8])$Counts))
	binned<-tmpbed %>% group_by(Name) %>% mutate(BIN = cut_number(Pos, 100, seq(1,100))) %>% ungroup() %>% group_by(BIN) %>% summarise_at("Counts", mean) %>% ungroup()
	binned$rep <- rep(q)
	if(grepl("WT", q)){
		binned$GT<-rep('WT')
	}else{
		binned$GT<-rep('Ski7')
	}
	FinalTab <- rbind(FinalTab, binned)
}
maxlim<-round(max(FinalTab$Counts))+1
x=ggplot(ungroup(FinalTab), aes(as.double(BIN), Counts, group = rep, col = GT)) + geom_line(stat = "identity") + scale_y_continuous(limits = c(0,maxlim)) + labs(title = paste(j, "in", i), x="% length")
NAME=paste0(paste(maindir, 'MetagenesBED6', i, j, "Graphs", sep = '/'), "Metagenes.pdf")
ggsave(NAME, x)
```
