# Analysis of transcript features of DEGs

After performing differential gene expression analysis with DESeq2 in R, the resulting up and downregulated genes were subjected to analyses to identify pontential inherent signatures.
For all the analyses below, only the longest transcripts (based on Ensembl release 92) were considered.


## Transcript lengths

A custom file containing the transcript, CDS, 5' UTR, and 3' UTR lengths (based on Ensembl 92 annotation) was generated.

```r
maindir <- "/"
samp<- c("Oocytes", "Eggs", "Embryos")

for(i in samp){
  if(i == "Eggs"){ tps <- c("INA", "ACT", "FER", "TPs") } else { tps <- c(paste0("T", 1:4), "TPs") }
  for(j in tps){  
    up<-read.table(paste0(maindir, 'Up_', i, '_DEGs_alph01_', j, '.txt'), header = T, stringsAsFactors = F, sep = '\t')[,1]
    down<-read.table(paste0(maindir, 'Down_', i, '_DEGs_alph01_', j, '.txt'), header = T, stringsAsFactors = F, sep = '\t')[,1]
    unc<-read.table(paste0(maindir, 'Unchanged_', i, '_DEGs_alph01_', j, '.txt'), header = T, stringsAsFactors = F, sep = '\t')[,1]
    for(p in c("Transcript", "CDS", "5UTR", "3UTR")){
      len<-read.table(paste0(maindir, '/', p, '_length.txt'), sep = '\t', stringsAsFactors = F, row.names = 1, header = T)
      upregions<-data.frame(DEGs = rep("Up"), Length = na.omit(len[paste0("gene:", up),2]), Alph = rep("A"))
      downregions<-data.frame(DEGs = rep("Down"), Length = na.omit(len[paste0("gene:", down),2]), Alph = rep("A"))
      uncregions<-data.frame(DEGs = rep("Unchanged"), Length = na.omit(len[paste0("gene:", unc),2]), Alph = rep("B"))
      all<-rbind(upregions, downregions, uncregions)
      mu<-ddply(all, 'DEGs', summarise, grp.mean = mean(log10(Length)))
      allecdf<-ddply(all, .(DEGs), transform, ECD = ecdf(Length)(Length))
      upks<-ks.test(upregions$Length, uncregions$Length, alternative = "two.sided")$p.value
      downks<-ks.test(downregions$Length, uncregions$Length, alternative = "two.sided")$p.value
      degs_ecdf<-ggplot(allecdf, aes(log10(Length), ECD)) + geom_line(aes(group = DEGs, col = DEGs), size = 1.5) + scale_colour_manual(values = c("darkorange", "deepskyblue3", "grey45")) + annotate("text", x= min(log10(allecdf$Length))+0.3, y = c(0.98, 0.95), label = c(paste("p = ", upks), paste("p = ", downks)), col = c("darkorange","deepskyblue3"))
      print(degs_ecdf)
    }
  }
}
```

## CAI analyses

The CAI analyses were performed with the perl script from [P Puigbo, _et al_, 2008](https://biologydirect.biomedcentral.com/articles/10.1186/1745-6150-3-38), and a zebrafish codon usage table
In addition, the GRCz11 fasta file, and the BED files split into up, down, or unchanged genes are required.

### Software requirements

| Software | Version |
|---|---|
| Bedtools | 2.27.1 |

```bash
bedtools getfasta -fi <GENOMEfasta> -bed <BED> -s -split -name -fo <OUTfa
start=4
trimend=3
awk -v start=$start -v trim=$trimend -v trx=trans '$0~trans{getline seq; end=int(length(seq)/3)*3; print; print substr(seq, start, end-3-trim)}' <OUTfa> > <OUTtmpfa>
num=($(grep -c ">" <OUTtmpfa>))
if [ $num -gt 1000 ]
then
  num=500
fi
len=($(awk '/^>/{if (l!="") print l; l=0; next}{l+=length($0)}END{print l}' <OUTtmpfa> | awk '{ total += $1; count++ } END { print int (total/count)+1 }'))
len=($(expr $len / 3))
perl CAIcal_ECAI_v1.4.pl -e cai_and_expected -f <OUTtmpfa> -h <CodonTable> -g 1 -o1 <OUTCAItxt> -o2 <OUTrandomtxt> -o3 <OUTexpectedtxt> -n $num -l $len -s y
```

Plot the ECDF of the CAI from DEGs
```r
upfile = #Path_to_CAIfile of up genes
downfile = #Path_to_CAIfile of down genes
uncfile = #Path_to_CAIfile of unchanged genes
outdir = #Path to output directory
prefix = #Prefix name for generated files

library(ggplot2)
library(ggpubr)
library(plyr)
library(DescTools)

up<-read.table(upfile, header = T, stringsAsFactors = F, sep = '\t')
down<-read.table(downfile, header = T, stringsAsFactors = F, sep = '\t')
unc<-read.table(uncfile, header = T, stringsAsFactors = F, sep = '\t')
up$DEGs<-rep("Up")
down$DEGs<-rep("Down")
unc$DEGs<-rep("Unchanged")
up$ECD <- ecdf(up$CAI)(up$CAI)
down$ECD <- ecdf(down$CAI)(down$CAI)
unc$ECD <- ecdf(unc$CAI)(unc$CAI)
up$Alph <- rep("A")
down$Alph <- rep("A")
unc$Alph <- rep("B")
all<-rbind(up, down, unc)
all$aECD<-ecdf(all$CAI)(all$CAI)
all$DEGs<-factor(all$DEGs, level = c("Up", "Down", "Unchanged"))
mu<-ddply(all, 'DEGs', summarise, grp.mean = mean(CAI))

upks<-ks.test(up$CAI, unc$CAI, alternative = "two.sided")$p.value
downks<-ks.test(down$CAI, unc$CAI, alternative = "two.sided")$p.value
degs_ecdf<-ggplot(all, aes(CAI, ECD)) + geom_line(aes(group = DEGs, col = DEGs), size = 1.5) + scale_colour_manual(values = c("darkorange", "deepskyblue3", "grey45")) + annotate("text", x= min(all$CAI)+0.03, y = c(0.98, 0.95), label = c(paste("p = ", upks), paste("p = ", downks)), col = c("darkorange","deepskyblue3")) + coord_cartesian(xlim = c(0.7, 0.875))
print(degs_ecdf)
```

## Read density over gene body regions

Files required: aligned BAM and the GTF file from Ensembl 92 release

### Software requirements

|Software|Version|
| --- | --- |
|python|2.7.15|
|HTseq|0.11.2|

```bash
samples= #list all the samples to be anlaysed
regions=('five_prime_UTR' 'three_prime_UTR' 'CDS')

for a in ${samples[@]}; do
  for r in ${regions[@]}; do
    if [ $r == 'five_prime_UTR' ]; then gbr=('5UTR'); elif [ $r == 'three_prime_UTR' ]; then gbr=('3UTR'); else gbr=('CDS'); fi;
	  htseq-count -f bam -r pos -s reverse -t $r -i Parent -m intersection-nonempty <OUTuniqbam> <GTF> > <OUTht_$gbr>
  done
done
```

To obtain the ratio of ski7/WT per gene body region, please refer to the script _Read_density_regions.r_
